src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+
+    callPhaseBlockDetector.getAll.foreach { block => callBlockLengthCounter.count(block.length) }
+    truthPhaseBlockDetector.getAll.foreach { block => truthBlockLengthCounter.count(block.length) }
+
+    val callBlockLengthMetrics  = callBlockLengthCounter.map { case (length, count) => new PhaseBlockLengthMetric(is_truth=false, length=length, count=count) }
+    val truthBlockLengthMetrics = truthBlockLengthCounter.map { case (length, count) => new PhaseBlockLengthMetric(is_truth=true, length=length, count=count) }
+    val blockLengthMetrics      = (callBlockLengthMetrics ++ truthBlockLengthMetrics).toSeq
+    
+    val callAssemblyStats  = AssemblyStatistics(callBlockLengthCounter)
+    val truthAssemblyStats = AssemblyStatistics(truthBlockLengthCounter)
+
+    metric.mean_called_block_length   = callBlockLengthCounter.mean()
+    metric.median_called_block_length = callBlockLengthCounter.median()
+    metric.stddev_called_block_length = callBlockLengthCounter.stddev(m=metric.mean_called_block_length)
+    metric.n50_called_block_length    = callAssemblyStats.n50
+    metric.n90_called_block_length    = callAssemblyStats.n90
+    metric.l50_called_block_length    = callAssemblyStats.l50
+    metric.mean_truth_block_length    = truthBlockLengthCounter.mean()
+    metric.median_truth_block_length  = truthBlockLengthCounter.median()
+    metric.stddev_truth_block_length  = truthBlockLengthCounter.stddev(m=metric.mean_truth_block_length)
+    metric.n50_truth_block_length     = truthAssemblyStats.n50
+    metric.n90_truth_block_length     = truthAssemblyStats.n90
+    metric.l50_truth_block_length     = truthAssemblyStats.l50
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // Output the metrics and finish up
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    logger.info("Outputting")
+
+    Metric.write(metrics=Seq(metric), path=PathUtil.pathTo(output + AssessPhasingMetric.MetricExtension))
+    Metric.write(metrics=blockLengthMetrics, path=PathUtil.pathTo(output + PhaseBlockLengthMetric.MetricExtension))
+
+    callReader.safelyClose()
+    truthReader.safelyClose()
+  }
+
+  private def callIterator(callReader: VCFFileReader): Iterator[VariantContext] = {
+    import scala.collection.JavaConversions.asScalaIterator
+    callReader.iterator.filter(v => v.isSNP && v.isBiallelic)
+  }
+
+  private def truthIterator(truthReader: VCFFileReader, intervalList: Option[IntervalList]): Iterator[VariantContext] = {
+    (intervalList match {
+      case Some(intv) => ByIntervalListVariantContextIterator(truthReader, intv, useIndex=false)
+      case None =>
+        import scala.collection.JavaConversions.asScalaIterator
+        truthReader.iterator.toIterator
+    }).filter(v => v.isSNP && v.isBiallelic)
+  }
+}
+
+private object AssemblyStatistics {
+  def apply(counter: NumericCounter[Long]): AssemblyStatistics = {
```

Comment:
It would help readability to either call the parameter something more specific (e.g. blockLengthCounter) or typedef Long -> BlockLength.